knowledge-base/grid-inline-edit-form.md

```diff
@@ -0,0 +1,213 @@
+
+    private List<SampleModel> GridData { get; set; } = new();
+
+    private TelerikGrid<SampleModel> GridRef { get; set; } = null!;
+
+    private SampleModel? GridItemInEditMode { get; set; }
+
+    private async Task OnGridEditButtonClick(SampleModel item)
+    {
+        GridItemInEditMode = item.Clone();
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>() { item };
+
+        await GridRef.SetStateAsync(gridState);
+    }
+
+    private async Task OnGridAddButtonClick()
+    {
+        var newItem = new SampleModel();
+        GridData.Insert(0, newItem);
+        GridRef.Rebind();
+
+        await OnGridEditButtonClick(newItem);
+    }
+
+    private async Task OnGridFormValidSubmit()
+    {
+        if (GridItemInEditMode == null)
+        {
+            return;
+        }
+
+        var originalItemIndex = GridData.FindIndex(x => x.Id == GridItemInEditMode!.Id);
+
+        if (GridItemInEditMode!.Id == 0)
+        {
+            GridItemInEditMode.Id = ++LastId;
+        }
+
+        GridData[originalItemIndex] = GridItemInEditMode!;
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>();
+
+        await GridRef.SetStateAsync(gridState);
+
+        GridItemInEditMode = null;
+    }
+
+    private async Task OnGridFormCancel()
+    {
+        if (GridItemInEditMode?.Id == 0)
+        {
+            var itemIndex = GridData.FindIndex(x => x.Id == 0);
+            GridData.RemoveAt(itemIndex);
+        }
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>();
+
+        await GridRef.SetStateAsync(gridState);
+
+        GridItemInEditMode = null;
+    }
+
+    private int LastId { get; set; }
+
+    protected override void OnInitialized()
+    {
+        var rnd = new Random();
+
+        for (int i = 1; i <= 5; i++)
+        {
+            GridData.Add(new SampleModel()
+                {
+                    Id = ++LastId,
+                    Name = $"Name {LastId}",
+                    Price = rnd.Next(1, 100) * 1.23m,
+                    Quantity = rnd.Next(0, 1000),
+                    StartDate = DateTime.Today.AddDays(-rnd.Next(1, 30)).AddMonths(-rnd.Next(1, 100))
+                });
+        }
+    }
+
+    public class SampleModel
+    {
+        [Display(AutoGenerateField = false)]
+        public int Id { get; set; }
+        [Required]
+        public string Name { get; set; } = string.Empty;
+        public decimal Price { get; set; }
+        public int Quantity { get; set; }
+        public DateTime? StartDate { get; set; }
+
+        public SampleModel Clone()
+        {
+            return new SampleModel()
+                {
+                    Id = Id,
+                    Name = Name,
+                    Price = Price,
+                    Quantity = Quantity,
+                    StartDate = StartDate
+                };
+        }
+    }
+}
+````
```

Comment:
Create a `## See Also` section at the bottom. The bare minimum of links is to the Grid and Form overviews and to their demos (the demos should open in a new window with the `<a href="" target="_blank"></a>` syntax. I would also add a link to the Inline Editing page.

Reply:
This KB article does not use the built-in editing mechanism of the Grid. So linking Inline Editing docs/demos would be confusing to the user.